stock_location_code/model/stock.py

```diff
@@ -65,21 +65,44 @@ def name_search(self, name, args=None, operator='ilike', limit=100):
             name, args=args, operator=operator, limit=limit)
         return res
 
-    @api.model
-    def _name_get(self, location):
+    @api.multi
+    def name_get(self):
         """Implements the stock location code in a new feauture, where if the
         location has a reference to a warehouse, the name in a m2o search
         concatenates the code and warehouse to a location if they exist.
         Visually, it's better to know which owns the warehouse location.
         """
-        barcode = "[%(barcode)s]" % {'barcode': location.loc_barcode} \
-            if location.loc_barcode else ''
-        # TODO: Add fields.function to get warehouse. get_warehouse is too slow
-        # TODO: Add a parameters for a location path name or a wh_name
-        warehouse = self.env['stock.warehouse'].browse(
-            self.get_warehouse(location))
-        wh_name = "(%(warehouse)s)" % {'warehouse': warehouse.name} \
-            if warehouse else ''
-        items = [barcode.strip(), location.name.strip(), wh_name.strip()]
-        new_name = ' '.join(item for item in items if item)
-        return new_name
+        res = []
+        if not self._ids:
+            return res
+        wh_obj = self.env['stock.warehouse']
+        wh_dict = {}.fromkeys(self._ids, False)
+        query = self._cr.mogrify('''
```

Comment:
What about use directly the execute (without mogrify) because supports this case?

Reply:
I just wanted to have the query to be executed prior to execution only for debugging purposes, though having the mogrify in there is not wrong either.

Reply:
you can have the query using print cr.query avoiding use mogrify